sdk/core/azure-core/azure/core/pipeline/policies/_universal.py

```diff
@@ -371,7 +371,7 @@ def __init__(self, logger=None, **kwargs):  # pylint: disable=unused-argument
             "azure.core.pipeline.policies.http_logging_policy"
         )
         self.allowed_query_params = set()
-        self.allowed_header_names = set(HttpLoggingPolicy.DEFAULT_HEADERS_WHITELIST)
+        self.allowed_header_names = set(self.DEFAULT_HEADERS_WHITELIST)
```

Comment:
Why? This is a class attribute, so it was correct

Reply:
I changed that to `self` because having it as HttpLoggingPolicy caused a bug in ARMHttpLoggingPolicy.

In `ARMHttpLoggingPolicy`, we set the DEFAULT_HEADERS_WHITELIST to be `HttpLoggingPolicy`'s, plus a couple ARM specific headers (code [here](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-mgmt-core/azure/mgmt/core/policies/__init__.py#L35)). However, by calling `HttpLoggingPolicy.DEFAULT_HEADERS_WHITELIST` here, we were setting the `allowed_header_names` of a `ARMHttpLoggingPolicy` to be just the `DEFAULT_HEADERS_WHITELIST` of `HttpLoggingPolicy`, not the `DEFAULT_HEADERS_WHITELIST` of `ARMHttpLoggingPolicy`. 

If you would like to still keep the code here as `HttpLoggingPolicy.DEFAULT_HEADERS_WHITELIST`, I can change the code in `ARMHttpLoggingPolicy` to override the setting of `self.allowed_header_names` to `ARMHttpLoggingPolicy.DEFAULT_HEADERS_WHITELIST`, I just went with this route because it was less code. Let me know @lmazuel !

Reply:
Ok, I get it now. So the correct fix is:
```
self.allowed_header_names = set(self.__class__.DEFAULT_HEADERS_WHITELIST)
```

Which means "use the current class attribute list to initialize yourself". `self` means the instance attributes, which is stretch here.